src/snowflake/snowpark/modin/pandas/groupby.py

```diff
@@ -736,8 +736,30 @@ def all(self, skipna=True):
         ErrorMessage.method_not_implemented_error(name="all", class_="GroupBy")
 
     def size(self):
-        # TODO: SNOW-1063349: Modin upgrade - modin.pandas.groupby.DataFrameGroupBy functions
-        ErrorMessage.method_not_implemented_error(name="size", class_="GroupBy")
+        if self._axis == 1:
+            return DataFrameGroupBy(
+                self._df.T.iloc[:, [0]],
+                self._by,
+                0,
+                drop=self._drop,
```

Comment:
In our previous discussion when talking about upstreaming, i think we agreed that the frontend will do basic parameter check,  conversion, and result correction. 
if that is the case, then it is probably better to move this just to query backend, you can call transform on it. If we are worry about axis = 1, I think that is a parameter scheduled to be deprecated anyway, we can improve our error message to ask customer to do the transformation before they call size also.

Reply:
This code is pulled directly from Modin OSS. The primary goal is to make as few changes as possible that need to be upstreamed. I don't think we should worry about `axis=1` use case.

Reply:
i think our primary goal is to make sure both Snowpark pandas and modin aligned on the frontend based on our previous architecture discussion, and the second goal is to minimize the work.

If we are not worried about axis = 1, how about remove this axis = 1 handling at frontend, and the backend check should throw an error for axis = 1 now.

Reply:
Removed axis=1, but I also put back our TODO since it's not aligned anymore.